tests/bluetooth/bsim/nrf_auraconfig/tester/src/nrf_auraconfig_test.c

```diff
@@ -0,0 +1,1164 @@
+#define PA_SYNC_SKIP			  5
+
+static K_SEM_DEFINE(sem_broadcast_sink_stopped, 0U, 1U);
+static K_SEM_DEFINE(sem_connected, 0U, 1U);
+static K_SEM_DEFINE(sem_disconnected, 0U, 1U);
+static K_SEM_DEFINE(sem_broadcaster_found, 0U, 1U);
+static K_SEM_DEFINE(sem_pa_synced, 0U, 1U);
+static K_SEM_DEFINE(sem_base_received, 0U, 1U);
+static K_SEM_DEFINE(sem_syncable, 0U, 1U);
+static K_SEM_DEFINE(sem_pa_sync_lost, 0U, 1U);
+static K_SEM_DEFINE(sem_broadcast_code_received, 0U, 1U);
+static K_SEM_DEFINE(sem_pa_request, 0U, 1U);
+static K_SEM_DEFINE(sem_past_request, 0U, 1U);
+static K_SEM_DEFINE(sem_bis_sync_requested, 0U, 1U);
+static K_SEM_DEFINE(sem_stream_connected, 0U, CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT);
+static K_SEM_DEFINE(sem_stream_started, 0U, CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT);
+static K_SEM_DEFINE(sem_big_synced, 0U, 1U);
+
+/* Sample assumes that we only have a single Scan Delegator receive state */
+static struct bt_bap_broadcast_sink *broadcast_sink;
+static struct bt_le_scan_recv_info broadcaster_info;
+static bt_addr_le_t broadcaster_addr;
+static struct bt_le_per_adv_sync *pa_sync;
+static uint32_t broadcaster_broadcast_id;
+static struct broadcast_sink_stream {
+	struct bt_bap_stream stream;
+} streams[CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT];
+
+static struct bt_bap_stream *streams_p[ARRAY_SIZE(streams)];
+static volatile bool big_synced;
+static volatile bool base_received;
+static struct bt_conn *broadcast_assistant_conn;
+static volatile uint8_t stream_count;
+
+extern enum bst_result_t bst_result;
+
+static struct nac_test_values_subgroup
+	subgroups_expected[CONFIG_BT_BAP_BROADCAST_SNK_SUBGROUP_COUNT];
+static struct nac_test_values_subgroup subgroups_dut[CONFIG_BT_BAP_BROADCAST_SNK_SUBGROUP_COUNT];
+
+static struct nac_test_values_big expected = {
+	.subgroups = subgroups_expected,
+};
+static struct nac_test_values_big dut = {
+	.subgroups = subgroups_dut,
+};
+
+static const struct bt_audio_codec_cap codec_cap = BT_AUDIO_CODEC_CAP_LC3(
+	BT_AUDIO_CODEC_CAP_FREQ_16KHZ | BT_AUDIO_CODEC_CAP_FREQ_24KHZ |
+		BT_AUDIO_CODEC_CAP_FREQ_48KHZ,
+	BT_AUDIO_CODEC_CAP_DURATION_10, BT_AUDIO_CODEC_CAP_CHAN_COUNT_SUPPORT(1), 40u, 120u, 1u,
+	BT_AUDIO_CONTEXT_TYPE_ANY);
+
+/* Create a mask for the maximum BIS we can sync to using the number of streams
+ * we have. We add an additional 1 since the bis indexes start from 1 and not
+ * 0.
+ */
+static const uint32_t bis_index_mask = BIT_MASK(ARRAY_SIZE(streams) + 1U);
+static uint32_t requested_bis_sync;
+static uint32_t bis_index_bitfield;
+static uint8_t sink_broadcast_code[BT_AUDIO_BROADCAST_CODE_SIZE];
+
+static char broadcast_name_target[ADV_NAME_MAX + 1];
+
+uint64_t total_rx_iso_packet_count; /* This value is exposed to test code */
+
+static void stream_connected_cb(struct bt_bap_stream *stream)
+{
+	LOG_DBG("Stream connected");
+
+	k_sem_give(&sem_stream_connected);
+}
+
+static void stream_disconnected_cb(struct bt_bap_stream *stream, uint8_t reason)
```

Comment:
I don't think that would be good in case we would want to connect to two different BIGs in one device. Not sure how we would do it practically, but the action of disconnecting shouldn't fail the test.